src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs

```diff
@@ -859,6 +911,99 @@ private static void OnPingComplete(object sender, PingCompletedEventArgs e)
             ((TestConnectionCommand)e.UserState)._pingComplete.Set();
         }
 
+        /// <summary>
+        /// The class contains information about the TCP connection test
+        /// </summary>
+        public class TcpTestStatus
+        {
+            /// <summary>
+            /// Initializes a new instance of the <see cref="TcpTestStatus"/> class.
+            /// </summary>
+            /// <param name="testNum">The number of this test.</param>
+            /// <param name="source">The source machine name or IP of the test.</param>
+            /// <param name="destination">The destination machine name or IP of the test.</param>
+            /// <param name="destinationAddress">The resolved IP from the destination</param>
+            /// <param name="port">The port used for the connection.</param>
+            /// <param name="latency">The latency of the test.</param>
+            /// <param name="result">The result of the test.</param>
+            internal TcpTestStatus(int testNum, string source, string destination, string destinationAddress, int port, long latency, TcpConnectionTestResult result)
+            {
+                TestNum = testNum;
+                Source = source;
+                Destination = destination;
+                DestinationAddress = destinationAddress;
+                Port = port;
+                Latency = latency;
+                Result = result;
+            }
+
+            /// <summary>
+            /// Gets and sets the count of the test.
+            /// </summary>
+            public int TestNum { get; set; }
+
+            /// <summary>
+            /// Gets the source from which the test was sent.
+            /// </summary>
+            public string Source { get; }
+
+            /// <summary>
+            /// Gets the destination name.
+            /// </summary>
+            public string Destination { get; }
```

Comment:
Please follow PingStatus class pattern and use Target, TargetAddress (with IPAddress type).

Reply:
Updated! 😊